File: 08_mar/LL_08_mar/LL_08_mar.py
class SList:
	def __init__(self):
		self.head = None

	def addFront(self, val):
		tempNode = Node(val)
		tempNode.next = self.head
		self.head = tempNode

	def printList(self):
		currNode = self.head
		lst = '[ '
		while currNode:
			lst += `currNode.value` + ', '
			currNode = currNode.next
		lst += ']'
		print lst

	def getLength(self):
		currNode = self.head
		counter = 0
		while currNode:
			counter += 1
			currNode = currNode.next
		return counter

	def getMax(self):
		if not self.head:
			return 'no list'
		currNode = self.head.next
		maximum = self.head.value
		while currNode:
			if currNode.value > maximum:
				maximum = currNode.value
			currNode = currNode.next
		return maximum

	def getMin(self):
		if not self.head:
			return 'no list'
		currNode = self.head.next
		minimum = self.head.value
		while currNode:
			if currNode.value < minimum:
				minimum = currNode.value
			currNode = currNode.next
		return minimum

	def getAvg(self):
		if not self.head:
			return 'no list'
		length = 1
		currNode = self.head.next
		avg = self.head.value
		while currNode:
			avg += currNode.value
			length += 1
			currNode = currNode.next
		return avg / length

class Node:
	def __init__(self, val):
		self.value = val
		self.next = None


myList = SList()

myList.addFront(8)
myList.addFront(6)
myList.addFront(7)
myList.addFront(5)
myList.addFront(3)
myList.addFront(0)
myList.addFront(9)

myList.printList()

print myList.getAvg()

File: js/08_mar/LL_08_mar/LL_08_mar.js
class Node {
	constructor(val) {
		this.value = val;
		this.next = null;
	}
}

class SList {
	constructor() {
		this.head = null;
	}

	addFront(val) {
		let tempNode = new Node(val);
		tempNode.next = this.head;
		this.head = tempNode;
	}

	printList() {
		let currNode = this.head,
			arr = '[ ';

		while (currNode) {
			arr += currNode.value + ', ';
			currNode = currNode.next;
		}

		arr += ']';

		console.log(arr);
	}

	getLength() {
		let currNode = this.head,
			counter = 0;

		while (currNode) {
			counter++;
			currNode = currNode.next;
		}

		return counter;
	}

	getMax() {
		if (!this.head) {
			return 'empty list';
		}

		let currNode = this.head.next,
			max = this.head.value;

		while (currNode) {
			if (currNode.value > max) {
				max = currNode.value;
			}

			currNode = currNode.next;
		}

		return max;
	}

	getMin() {
		if (!this.head) {
			return 'empty list';
		}

		let currNode = this.head.next,
			min = this.head.value;

		while (currNode) {
			if (currNode.value < min) {
				min = currNode.value;
			}

			currNode = currNode.next;
		}

		return min;
	}

	getAvg() {
		if (!this.head) {
			return 'empty list';
		}

		let currNode = this.head.next,
			avg = this.head.value,
			length = 1;

		while (currNode) {
			avg += currNode.value;
			length++;
			currNode = currNode.next;
		}

		return avg / length;
	}
}

var myList = new SList();

myList.addFront(8);
myList.addFront(6);
myList.addFront(7);
myList.addFront(5);
myList.addFront(3);
myList.addFront(0);
myList.addFront(9);

myList.printList();

console.log(myList.getAvg());
